Neuro/Utils/Methods.cs

```diff
@@ -5,47 +5,43 @@ namespace Neuro.Utils;
 
 public static class Methods
 {
-    private static Dictionary<Vector2, string> locationNames = new Dictionary<Vector2, string>()
-    {
-        { new Vector2 { x = 0, y = 0 }, "Cafeteria" },
-        { new Vector2 { x = 9, y = 1 }, "Weapons" },
-        { new Vector2 { x = 6.6f, y = -3.7f }, "O2" },
-        { new Vector2 { x = 17, y = -5 }, "Navigation" },
-        { new Vector2 { x = 9, y = -12.5f }, "Shields" },
-        { new Vector2 { x = 4, y = 15.6f }, "Communications" },
-        { new Vector2 { x = 0, y = -17 }, "Trash Chute" },
-        { new Vector2 { x = 0, y = -12 }, "Storage" },
-        { new Vector2 { x = -9, y = -10f }, "Electrical" },
-        { new Vector2 { x = -15.6f, y = -10.6f }, "Lower Engine" },
-        { new Vector2 { x = -13f, y = -4.4f }, "Security" },
-        { new Vector2 { x = -21f, y = -5.5f }, "Reactor" },
-        { new Vector2 { x = -15.4f, y = 1 }, "Upper Engine" },
-        { new Vector2 { x = -8f, y = -3.5f }, "MedBay" },
-        { new Vector2 { x = 5f, y = -8 }, "Admin" },
-    };
-
     public static string GetLocationFromPosition(Vector2 position)
     {
         float closestDistance = Mathf.Infinity;
-        string closestLocation = "";
+        PlainShipRoom closestLocation = null;
+        string nearPrefix = "outside near "; // If we're not in any rooms/hallways, we're "outside"
+
+        if (ShipStatus.Instance == null) // In case this is called from the lobby
+            return "the lobby";
 
-        foreach (KeyValuePair<Vector2, string> keyValuePair in locationNames)
+        foreach (PlainShipRoom room in ShipStatus.Instance.AllRooms)
         {
-            float distance = Vector2.Distance(keyValuePair.Key, position);
-            if (distance < 2f)
+            Collider2D collider = room.roomArea;
+            if (collider.OverlapPoint(position))
```

Comment:
You are only ignoring the current room when performing this check, if the current room is Hallway and the nearest room is also a Hallway, then this code will produce "a hallway near Hallway".

You should ignore all rooms with a systemtype of Outside or Hallway

Reply:
Fixed, although I'm not sure if there's any actual locations where this would happen. As I said in other comments, there's no need to check for Outside here since that doesn't appear in the rooms list, but I've excluded hallways.

Reply:
This point is still valid though, is it not?